File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>
#define n_div 1234567

using namespace std;

// int fibo(int n){
    // if(n<=0) return 0;
    // if(n == 1) return 1;
    // return fibo(n-1)%n_div + fibo(n-2)%n_div;
// }

int solution(int n) {
    // int answer = fibo(n)%n_div;
    vector<int> fibo;
    
    for(int i=0; i<=n; i++){
        if(i<2) fibo.push_back(i);
        else fibo.push_back(fibo[i-2]%n_div + fibo[i-1]%n_div);
    }
    
    int answer = fibo[n]%n_div;
    return answer;
}

File: 프로그래머스/lv2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    fibo = []
    for i in range(0,n+1):
        if i<2:
            fibo.append(i)
        else:
            fibo.append(fibo[i-1]%1234567 + fibo[i-2]%1234567)
            
    answer = fibo[n]%1234567
    return answer